src/main/java/seedu/address/logic/commands/FindBandCommand.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.band.Band;
+
+/**
+ * Lists all musicians in a band input by the user.
+ */
+public class FindBandCommand extends Command {
+
+    public static final String COMMAND_WORD = "findb";
+
+    public static final Object MESSAGE_USAGE = COMMAND_WORD + ": Finds all musicians who are members of "
+        + "the band with a specific band name and displays them as a list with index numbers.\n"
+        + "Parameters: BAND NAME\n"
+        + "Example: " + COMMAND_WORD + " BlackPink";
+
+    private final Predicate<Band> predicate;
+
+    public FindBandCommand(Predicate<Band> predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        model.updateFilteredBandMusicianList(predicate);
+
+        if (model.getFilteredBandList().size() > 1) {
+            throw new CommandException(Messages.MESSAGE_UNKNOWN_BAND);
```

Comment:
It seems like this part of the code is restricting the number of bands fetched from the `findb` command to 1. Anything larger than this will display an error. It could be deduced that either the command correctly executes and returns one single band, or fails to execute and returns all the band (which will then report errors in this part of the code), but the logic could probably be made clearer?

Reply:
Yes I agree with you that this logic is a bit counterintuitive right now, do you have any suggestions on how to make it clearer? Or do you mean adding a comment.